src/main/java/com/sourcegraph/cody/agent/CodyAgentClient.java

```diff
@@ -26,14 +33,112 @@ public class CodyAgentClient {
   // onSetConfigFeatures
   @Nullable public Consumer<WebviewPostMessageParams> onReceivedWebviewMessage;
 
-  @Nullable public Editor editor;
+  // Callback for the "editTask/didUpdate" notification from the agent.
+  @Nullable private Consumer<EditTask> onEditTaskDidUpdate;
+
+  // Callback for the "editTask/didDelete" notification from the agent.
+  @Nullable private Consumer<EditTask> onEditTaskDidDelete;
+
+  // Callback for the "textDocument/edit" request from the agent.
+  @Nullable private Consumer<TextDocumentEditParams> onTextDocumentEdit;
+
+  // Callback for the "workspace/executeCommand" request from the agent.
+  @Nullable private Consumer<DisplayCodeLensParams> onDisplayCodeLens;
+
+  // Callback for the "workspace/edit" request from the agent.
+  @Nullable private Consumer<WorkspaceEditParams> onWorkspaceEdit;
+
+  public void setOnEditTaskDidUpdate(@Nullable Consumer<EditTask> callback) {
+    onEditTaskDidUpdate = callback;
+  }
+
+  public void setOnEditTaskDidDelete(@Nullable Consumer<EditTask> callback) {
+    onEditTaskDidDelete = callback;
+  }
+
+  @JsonNotification("editTask/didUpdate")
+  public void editTaskDidUpdate(EditTask params) {
+    onEventThread(
+        () -> {
+          if (onEditTaskDidUpdate != null) {
```

Comment:
For brevity, what about providing a default callback which logs, making the callback non-nullable everywhere, and just calling directly here?